NoMongo: Restore the GraphQL Introspection GitHub Action Test #3485

Priyanshuthapliyal2005 opened the pull request:
<!--
This section can be deleted after reading.

We employ the following branching strategy to simplify the development process and to ensure that only stable code is pushed to the `master` branch:

- `develop`: For unstable code: New features and bug fixes.
- `master`: Where the stable production ready code lies. Only security related bugs.

NOTE!!!

ONLY SUBMIT PRS AGAINST OUR `DEVELOP` BRANCH. THE DEFAULT IS `MAIN`, SO YOU WILL HAVE TO MODIFY THIS BEFORE SUBMITTING YOUR PR FOR REVIEW. PRS MADE AGAINST `MAIN` WILL BE CLOSED.

-->

<!--
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request.
-->

**What kind of change does this PR introduce?**
NoMongo: Restore the GraphQL Introspection GitHub Action Test 
<!-- E.g. a bugfix, feature, refactoring, etc… -->

**Issue Number:**

Fixes #3485 <!--Add related issue number here.-->

**Snapshots/Videos:**

<!--Add snapshots or videos wherever possible.-->

**If relevant, did you update the documentation?**

<!--Add link to Talawa-Docs.-->

**Summary**

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
<!-- Try to link to an open issue for more information. -->

**Does this PR introduce a breaking change?**

<!-- If this PR introduces a breaking change, please describe the impact and a migration path for existing applications. -->

## Checklist

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [ ] I have written tests for all new changes/features
- [ ] I have verified that test coverage meets or exceeds 95%
- [ ] I have run the test suite locally and all tests pass


**Other information**

<!--Add extra information about this PR here-->

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

<!--Yes or No-->

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **Workflow Updates**
	- Refined GitHub Actions workflow configuration
	- Updated workflow triggers to target specific branch
	- Modified Node.js version from 22.x to 20.x
	- Added GraphQL schema validation job
	- Updated documentation deployment process

- **Dependencies**
	- Added GraphQL inspector CLI as a development dependency

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Priyanshuthapliyal2005 commented:
@palisadoes can you add 'ignore-sensitive-files-pr' label over this pr

Cioppolo14 commented:
@Priyanshuthapliyal2005 Please fix the issue you are reference, it currently references a closed PR.  Please do not assign reviewers and please fix all failed checks.

palisadoes commented:
1. Closing. 
2. Once again you have made wholesale changes to pull-request.yml which will cause the workflows to break.
3. In some cases you have changed the logic 
4. In other cases you have removed functionality
5. Please be more careful and responsible about the quality of our code

![image](https://github.com/user-attachments/assets/2f31e480-e373-4292-9a29-d2241e41d9b8)
![image](https://github.com/user-attachments/assets/0f32b3ac-f054-49a2-9876-99c99406e6ba)
![image](https://github.com/user-attachments/assets/8a28c7dc-2366-454c-b38e-d8a24a3a0cd2)
![image](https://github.com/user-attachments/assets/97f2ef46-e0b6-4c8d-a611-bb68ac7d6083)

Priyanshuthapliyal2005 commented:
I will review all the changes. I formatted the YML file, which caused changes throughout the code. I will take care of it this time.

palisadoes commented:
Yes, but you also removed and modified previous functionality

Priyanshuthapliyal2005 commented:
> Yes, but you also removed and modified previous functionality

Yeah , I have removed few dependency and functionality,
Like some python commands and graph ql inspector global dependency and many more

palisadoes commented:
That was out of scope of the issue. It jeopardized the repo. 

If you see functionality that needs fixing, then open an issue specific for that. 

Validating multiple features on one PR is not optimal especially when there is no explanation of context. 

Focus on the needs of the issue
